akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala

```diff
@@ -57,12 +57,12 @@ abstract class JournalSpec(config: Config) extends PluginSpec(config) {
 
   def writeMessages(fromSnr: Int, toSnr: Int, pid: String, sender: ActorRef, writerUuid: String): Unit = {
     val msgs =
-      if (supportsAtomicPersistAllOfSeveralEvents)
-        (fromSnr to toSnr).map { i ⇒
-          AtomicWrite(PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
-            writerUuid = writerUuid))
-        }
-      else
+      if (supportsAtomicPersistAllOfSeveralEvents) {
+        List(AtomicWrite((fromSnr to toSnr) map { i ⇒
+          PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
+            writerUuid = writerUuid)
+        }))
+      } else {
         (fromSnr to toSnr - 1).map { i ⇒
           if (i == toSnr - 1)
             AtomicWrite(List(
```

Comment:
There is some misunderstanding and it can be on my end. There should not be any AtomicWrite(List(1,2,3)) when `supportsAtomicPersistAllOfSeveralEvents == false`.

Why can't we just swap the cases?

    val msgs =
      if (supportsAtomicPersistAllOfSeveralEvents)
        (fromSnr to toSnr - 1).map { i ⇒
          if (i == toSnr - 1)
            AtomicWrite(List(
              PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
                writerUuid = writerUuid),
              PersistentRepr(payload = s"a-${i + 1}", sequenceNr = i + 1, persistenceId = pid, sender = sender,
                writerUuid = writerUuid)))
          else
            AtomicWrite(PersistentRepr(payload = s"a-${i}", sequenceNr = i, persistenceId = pid, sender = sender,
              writerUuid = writerUuid))
        }

      else
        (fromSnr to toSnr).map { i ⇒
          AtomicWrite(PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
            writerUuid = writerUuid))
        }


Note that in the `supportsAtomicPersistAllOfSeveralEvents == true` case I would like to have a mix of things, i.e.
List(AtomicWrite(1), AtomicWrite(2), AtomicWrite(3, 4))